clang/runtime/dpct-rt/include/device.hpp.inc

```diff
@@ -463,12 +463,25 @@ public:
   }
   device_ext(const sycl::device &base)
       : sycl::device(base), _ctx(*this) {
+#ifdef DPCT_PROFILING_ENABLED
```

Comment:
I think this code can be refactored to be more compact and less duplications.
For example you can define properties first with macro-related logic and then use one call without any macros, instead of repeating the call multiple times under different macros defined.
Also, consider moving it to some function, since there are 3 almost identical changes done - need to avoid duplication like that.

Reply:
OKay